gencap.f90

```diff
@@ -12,324 +12,468 @@
+    ! This is for use with MESA arrays specifically (though could likely be used for GARSTEC)
+    ! star initialization done elsewhere
+    ! subroutine captn_mesa(mx_in,sigma_0_in,niso_in,nq_in,nv_in,capped)
+    !   subroutine captn_mesa()
+    !   use capmod
+    !   implicit none
+    !   ! integer, intent(in):: nq_in, niso_in, nv_in
+    !   integer i, ri
+    !   ! double precision, intent(in) :: mx_in, sigma_0_in
+    !   double precision :: capped, maxcap !this is the output
+    !   double precision :: epsabs, epsrel,limit,result,abserr,neval !for integrator
+    !   double precision :: ier,alist,blist,rlist,elist,iord,last!for integrator
+    !   double precision, allocatable :: u_int_res(:)
+    !
+    !
+    ! end subroutine captn_mesa
+
+
+    subroutine captn_general(mx_in,sigma_0,niso,nq_in,nv_in,spin_in,capped)
+      use capmod
+      implicit none
+      integer, intent(in):: nq_in, nv_in, niso, spin_in
+      ! integer, intent(in):: spin_in
+      integer eli, ri, limit
+      double precision, intent(in) :: mx_in, sigma_0
+      double precision :: capped !this is the output
+      double precision :: sigma_SD, sigma_SI
+      double precision :: maxcap, maxcapped, a, muminus, sigma_N, umax, umin, vesc
+      double precision :: epsabs, epsrel, abserr, neval  !for integrator
+      double precision :: ier,alist,blist,rlist,elist,iord,last!for integrator
+      double precision, allocatable :: u_int_res(:)
+
+      dimension alist(1000),blist(1000),elist(1000),iord(1000),   rlist(1000)!for integrator
+      external integrand
+      external gausstest !this is just for testing
+
+      epsabs=1.d-8
+      epsrel=1.d-8
+      limit=1000
+
+      mdm = mx_in
+      nq = nq_in
+      nv = nv_in
+
+      if (spin_in == 1) then
+        sigma_SD = sigma_0
+        sigma_SI = 0.d0
+      else if (spin_in == 0) then
+        sigma_SD = 0.d0
+        sigma_SI = sigma_0
+      end if
 
-    subroutine captn_general(mx_in,sigma_0_in,niso_in,nq_in,nv_in,capped)
-    use capmod
-    implicit none
-    integer, intent(in):: nq_in, niso_in, nv_in
-    integer i, ri
-    double precision, intent(in) :: mx_in, sigma_0_in
-    double precision :: capped, maxcap !this is the output
-    double precision :: epsabs, epsrel,limit,result,abserr,neval !for integrator
-    double precision :: ier,alist,blist,rlist,elist,iord,last!for integrator
-    double precision, allocatable :: u_int_res(:)
-
-    dimension alist(1000),blist(1000),elist(1000),iord(1000),   rlist(1000)!for integrator
-    external gausstest !this is just for testing
-    external integrand
-    external dummyf
-    epsabs=1.d-17
-    epsrel=1.d-17
-    limit=1000
-
-
-    mdm = mx_in
-    sigma_0 = sigma_0_in
-    niso = niso_in
-    nq = nq_in
-    nv = nv_in
-
-
-
-    if (nq*nv .ne. 0) then
-    print*, "Oh no! nq and nv can't both be nonzero. "
-    return
-    end if
+      if (nq*nv .ne. 0) then
+      print*, "Oh no! nq and nv can't both be nonzero."
+      return
```

Comment:
Would this be cleaner as a 'stop' command like the error below?

Reply:
Yes